src/Core/IdentityServer/BaseRequestValidator.cs

```diff
@@ -259,11 +274,63 @@ protected async Task BuildErrorResultAsync(bool twoFactorRequest, T context, Use
             return new Tuple<bool, Organization>(individualRequired || firstEnabledOrg != null, firstEnabledOrg);
         }
 
+        private async Task<bool> RequiresSsoAuthAsync(User user, string grantType)
+        {
+            if (grantType == "authorization_code")
+            {
+                return true; // Already using SSO to authorize, finish successfully
+            }
+            
+            // Is user apart of any orgs? Use cache for initial checks.
+            var orgs = (await _currentContext.OrganizationMembershipAsync(_organizationUserRepository, user.Id))
```

Comment:
I know this pattern is established in `RequiresTwoFactorAsync(User user)`, however for future reference `.ToList()` is not required here since this method already returns an `ICollection<T>` and since it is not a more general `IEnumerable<T>` or `IQueryable<T>` it's safer to assume it is not "late bound" (where LINQ extensions could interfere with proper execution of DB queries or other I/O when enumerating).

Leave this as-is for now for consistency, just wanted to point this out.

Reply:
👍